File: backend/backend/src/main/java/com/bezkoder/springjwt/models/MaintenanceSettings.java
package com.bezkoder.springjwt.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MaintenanceSettings {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int optimizeIndexesOlderThanDays;
    private int closeIndexesOlderThanDays;
    private int deleteIndexesOlderThanDays;
    private String repositoryToStoreSnapshots;
    private String deleteSnapshotsOlderThan;

    private String settingName;
    private String settingValue;
    private String description;
    private String createdBy;
    private String createdDate;
    private String lastModifiedBy;
    private String lastModifiedDate;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "maintenance_settings_id")
    private List<Repository> repositories;
    @ManyToOne
    @JoinColumn(name = "cluster_id")
    private Cluster cluster;
    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }

}

File: frontend/font-end/main/src/app/models/MaintenanceSettings.ts
import { Repository } from "./Repository";

// src/app/models/maintenance-settings.model.ts
export class MaintenanceSettings {
  id?: number;
  optimizeIndexesOlderThanDays: number;
  closeIndexesOlderThanDays: number;
  deleteIndexesOlderThanDays: number;
  repositoryToStoreSnapshots: string;
  deleteSnapshotsOlderThan: string;
  repositories?: Repository[];
  settingName?: string;
  settingValue?: string;
  description?: string;
  createdBy?: string;
  createdDate?: string;
  lastModifiedBy?: string;
  lastModifiedDate?: string;
  clusterId?: number; // Ajoutez cette ligne si nécessaire
}
